src/Peering/Peering/PeeringService/Prefix/GetAzurePeeringServicePrefixCommand.cs

```diff
@@ -0,0 +1,191 @@
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+
+    using Microsoft.Azure.Commands.Peering.Properties;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.Management.Peering;
+    using Microsoft.Azure.Management.Peering.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Common;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Models;
+
+    /// <inheritdoc />
+    /// <summary>
+    ///     The Get Az InputObject Legacy peering.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzPeeringServicePrefix", SupportsShouldProcess = true, DefaultParameterSetName = Constants.ParameterSetNameByResourceAndName)]
+    [OutputType(typeof(PSPeeringServicePrefix))]
+    public class GetAzurePeeringServicePrefixCommand : PeeringBaseCmdlet
+    {
+        /// <summary>
+        /// Gets or sets the peering service input object
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.PrefixInputObjectHelp,
+            ValueFromPipeline = true,
+            ParameterSetName = Constants.ParameterSetNameDefault)]
+        [ValidateNotNullOrEmpty]
+        public PSPeeringService InputObject { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the ResourceGroupName
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceGroupNameHelp,
+            ParameterSetName = Constants.ParameterSetNameByResourceAndName)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the InputObject name.
+        /// </summary>
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringServiceHelp,
+            ParameterSetName = Constants.ParameterSetNameByResourceAndName)]
+        public string PeeringServiceName { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the InputObject name.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.ParameterSetNameByResourceAndName)]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.ParameterSetNameDefault)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The resource  id
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.ResourceIdHelp,
+            ParameterSetName = Constants.ParameterSetNameByResourceId)]
```

Comment:
@rileymckenna this parameter should have `ValueFromPipelineByPropertyName = true`